### DISCUSSION TITLE
New keyword: empty

### DISCUSSION THREAD

**Discussion by TheTrigger** [dotnet/csharplang#4435] [Language Ideas]
We already reached `is null` and `is not null`

Now what about `is [not] empty` for enumerables?

```
null is empty => true
"" is empty => true
```

I guess null == empty goes against some null conventions...

However, I don't like parentheses, they distract me, and I feel the `empty` word very uncomfortable to type... but...

**Comment by 333fred** [comment]
See https://github.com/dotnet/csharplang/issues/3435.

**Comment by alrz** [comment]
`string.IsNullOrEmpty` is `s is null or ""` or `s is null or { Length: 0 }`

With length patterns it would be `s is null or [0]` or `e is [>0]` instead of `e.Any()`

Length patterns were initially discussed in https://github.com/dotnet/csharplang/blob/master/meetings/2020/LDM-2020-12-16.md

**Reply by TonyValenti** [reply]
For enumerables I use an IsEmpty extension method(!Any) and for strings I use a IsBlank(string.IsNullOrEmpty) extension method to shortcut all that unnecessary typing.

**Comment by huoyaoyuan** [comment]
*New keyword* is very likely to be rejected.

**Reply by huoyaoyuan** [reply]
Generally, I think keyword can be introduced as a necessary part of a big set of feature. It shouldn't be introduced for a minor feature.
